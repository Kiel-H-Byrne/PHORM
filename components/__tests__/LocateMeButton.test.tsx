import { Map, initialize, mockInstances } from "@googlemaps/jest-mocks";
import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import FloatingButtons from "../LocateMeButton";

beforeEach(() => {
  initialize();
});

// Mock the necessary dependencies and props
// jest.mock('google.maps.Marker', () => jest.fn());
// jest.mock('google.maps.Circle', () => jest.fn());

const mockSetClientLocation = jest.fn(); // Mock the setClientLocation prop

describe("LocateMeButton", () => {
  it("should render the component correctly", () => {
    const { container } = render(
      <FloatingButtons
        mapInstance={mockInstances.get(Map)[0]}
        setClientLocation={mockSetClientLocation}
        clientLocation={null}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should call handleClick when the "My Location" button is clicked', () => {
    const { getByLabelText } = render(
      <FloatingButtons
        mapInstance={mockInstances.get(Map)[0]}
        setClientLocation={mockSetClientLocation}
        clientLocation={null}
      />
    );

    const myLocationButton = getByLabelText("My Location");
    fireEvent.click(myLocationButton);

    // Add your assertions here to check if the necessary functions have been called
    expect(mockSetClientLocation).toHaveBeenCalledTimes(1);
    // ...
  });

  it('should call handleOpen when the "Add Listing" button is clicked', () => {
    const { getByLabelText } = render(
      <FloatingButtons
        mapInstance={mockInstances.get(Map)[0]}
        setClientLocation={mockSetClientLocation}
        clientLocation={null}
      />
    );

    const addListingButton = getByLabelText("Add Listing");
    fireEvent.click(addListingButton);

    // Add your assertions here to check if the necessary functions have been called
    // ...
  });
});
