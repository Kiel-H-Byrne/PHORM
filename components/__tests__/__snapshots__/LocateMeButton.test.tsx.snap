// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocateMeButton should render the component correctly 1`] = `
<div>
  <button
    aria-label="My Location"
    class="chakra-button css-1rvxpru"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm8.94 3A8.994 8.994 0 0013 3.06V1h-2v2.06A8.994 8.994 0 003.06 11H1v2h2.06A8.994 8.994 0 0011 20.94V23h2v-2.06A8.994 8.994 0 0020.94 13H23v-2h-2.06zM12 19c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
      />
    </svg>
  </button>
  <button
    aria-label="Add PullUp"
    class="chakra-button css-r4q2dn"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 16c0 1.103.897 2 2 2h3.586L12 21.414 15.414 18H19c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2H5c-1.103 0-2 .897-2 2v12zM5 4h14v12h-4.414L12 18.586 9.414 16H5V4z"
      />
      <path
        d="M11 14h2v-3h3V9h-3V6h-2v3H8v2h3z"
      />
    </svg>
  </button>
</div>
`;
